src/index.js

```diff
@@ -252,6 +252,12 @@ Promise._unhandledRejectionFn = function _unhandledRejectionFn(err) {
   if (typeof console !== 'undefined' && console) {
     console.warn('Possible Unhandled Promise Rejection:', err); // eslint-disable-line no-console
   }
+  if (typeof window !== 'undefined' && window && typeof window.dispatchEvent === "function") {
+    const evt = new CustomEvent("unhandledrejection");
+    evt.reason = err;
+    evt.promise = this;
```

Comment:
I am unsure if `this` is a reference to the Promise object because `_unhandledRejectionFn` does not seem to be part of Promise.prototype.

Is there any reference to the Promise object available in this function?
If yes, then the reference should be assigned to `evt.promise`.
If no, then one might consider passing the reference to the function (see [here](https://developer.mozilla.org/en-US/docs/Web/API/Window/unhandledrejection_event)).